python/selfie-lib/selfie_lib/PerCharacterEscaper.py

```diff
@@ -1,54 +1,43 @@
+from typing import List
+
 class PerCharacterEscaper:
-    def __init__(self, escape_code_point, escaped_code_points, escaped_by_code_points):
-        self.escape_code_point = escape_code_point
-        self.escaped_code_points = escaped_code_points
-        self.escaped_by_code_points = escaped_by_code_points
+    def __init__(self, escape_code_point: int, escaped_code_points: List[int], escaped_by_code_points: List[int]):
+        self.__escape_code_point = escape_code_point
+        self.__escaped_code_points = escaped_code_points
+        self.__escaped_by_code_points = escaped_by_code_points
 
-    @staticmethod
-    def _first_offset_needing_escape(
-        input_string, escaped_code_points, escape_code_point=None
-    ):
+    def __first_offset_needing_escape(self, input_string: str, escape_code_point: int = None) -> int:
```

Comment:
Why is `escape_code_point` an argument at all? Can't we remove it and just use `__escape_code_point`? Seems like that's what we do on line 13-14 below anyway.

The kotlin doesn't need any argumens besides `input_string`.

https://github.com/diffplug/selfie-python-wip/blob/f8f51a4d2b1ac22a6df3f61353454b48b38e1c06/jvm/selfie-lib/src/jvmMain/kotlin/com/diffplug/selfie/PerCharacterEscaper.jvm.kt#L36-L51

Reply:
I have make the necessary changes to use __escape_code_point